src/components/Buttons/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,43 @@
+import { SetStateAction } from "react";
+import style from "./Checkbox.module.scss";
+import { RegisterCheckboxValue } from "@/services/types";
+
+interface ICheckbox {
+  label: string;
+  initialValue: RegisterCheckboxValue;
+  registerName: string;
+  setValue: (name: string, value: SetStateAction<RegisterCheckboxValue>) => void;
+  getValues: (name: string) => any;
+}
+
+const Checkbox = ({ label, initialValue, registerName, setValue, getValues }: ICheckbox) => {
```

Comment:
**nit:** Your prop interface should look like following to make it properly reusable:

```ts
interface ICheckbox {
  id: string;
  label: string;
  name?: string;
  value?: string;
  checked: boolean;
  onChange(e: React.ChangeEvent<HTMLInputElement>): void
}
```

Try to implement this - it will force you to refactor this component correctly